github_app.go

```diff
@@ -63,20 +67,19 @@ func getGithubAppAuth(s *Settings) *githubAppAuth {
 		return nil
 	}
 
-	return &githubAppAuth{
-		appId:      s.AuthConfig.Github.AppId,
-		privateKey: []byte(key),
+	return &GithubAppAuth{
+		AppID:      s.AuthConfig.Github.AppId,
+		PrivateKey: []byte(key),
 	}
 }
 
 // HasGitHubApp returns true if the GitHub app is installed on given owner/repo.
-func (s *Settings) HasGitHubApp(ctx context.Context, owner, repo string) (bool, error) {
-	authFields := getGithubAppAuth(s)
-	if authFields == nil {
-		return false, errors.New("GitHub app is not configured in admin settings")
+func HasGitHubApp(ctx context.Context, githubAppAuth *GithubAppAuth, owner, repo string) (bool, error) {
```

Comment:
With the changes and expanded use case, I think a different name, perhaps something like `IsGithubAppInstalledOnRepo` might be a better fit.